repo/packages/detray/package.py

```diff
@@ -0,0 +1,30 @@
+from spack.package import *
+
+
+class Detray(CMakePackage):
+    homepage = "https://github.com/acts-project/detray"
+    git = "https://github.com/acts-project/detray.git"
+    list_url = "https://github.com/acts-project/detray/tags"
+
+    version("0.27.0", commit="2e593264f031c2bf2430cb3a2e80f34e40fa5a29")
+    version("0.22.0", commit="727f6a9f65a5eb47731730d70460a0587b31fe0d")
+
+    depends_on("cmake@3.17:", type="build")
+    depends_on("vecmem")
+    depends_on("covfie")
+    depends_on("nlohmann-json")
+    depends_on("dfelibs")
+    depends_on("algebra-plugins +vecmem +vc +eigen")
+
+    patch("fix_cmake_install.patch", when="@0.22.0:", level=1)
+    patch("nlohmann_target.patch", when="@0.22.0:", level=1)
+
+    def cmake_args(self):
+        args = [
+            "-DCMAKE_BUILD_TYPE=Release",
```

Comment:
I would strongly advise against this.

1. Spack's CMake support already defines a `build_type` variant for you, you can use it to tweak this however you want without hardcoding it in your package.
2. By forcing this to `Release`, you break people's CPU profilers by default and force them into a full rebuild when they figure out they need to profile something, which is not nice. `RelWithDebInfo` is a better default because of this.

This applies to all of your other packages.